### PR TITLE
Clarifications to feature flag doc

### PR DESCRIPTION
**Pull request by trmartin4**
## Objective

Added a few clarifications to the feature flag doc:

- Added clarification on naming
- Added guidance on when to request flags in LaunchDarkly
- Removed the "technical debt" term from the cleanup tasks to emphasize how this should be a part of the feature and not something that we might get to eventually.

### CODE REVIEW

File: docs/contributing/featureflags.md [hunk 1]
```diff
@@ -91,7 +91,8 @@ Once you have decided that a feature flag is necessary, the first step is to dec
 Recommendations for naming are:
 
 - Name the flag using kebab-case (lowercase and dash-separated, such as `enable-feature`).
-- Name the flag in the affirmative when possible (`enable-feature`, not `disable-feature`).
+- For boolean flags, it is not necessary to include the `enable` verb, as it is implied by it being
```
Comment (withinfocus) [inline docs/contributing/featureflags.md:94]: ⛏️ I capitalize Boolean.

File: docs/contributing/featureflags.md [hunk 2]
```diff
@@ -124,7 +125,15 @@ the appropriate access. You should discuss:
 - The data type of the flag.
 - The default value of the flag.
 - The possible values of the flag (for non-boolean types).
-- Any context-based rules that should drive flag behavior.
+- Any context-based rules that should drive flag behavior. :::tip When should I request the flag in
```
Comment (withinfocus) [inline docs/contributing/featureflags.md:128]: ❓ Newline needed?

File: docs/contributing/featureflags.md [hunk 3]
```diff
@@ -124,7 +125,15 @@ the appropriate access. You should discuss:
...
+  LaunchDarkly?
+
+As a general rule, feature flag should be requested for creation in LaunchDarkly as part of merging
```
Comment (withinfocus) [inline docs/contributing/featureflags.md:131]: ⛏️ flag_s_.

File: docs/contributing/featureflags.md [hunk 4]
```diff
@@ -124,7 +125,15 @@ the appropriate access. You should discuss:
...
+the code using the flag into the `master` branch. Since local development and QA testing with their
+self-hosted instances will use local data sources, the first time that a flag in LaunchDarkly would
+be referenced is when the code is deployed to a cloud environment from the `master` branch.
```
Comment (withinfocus) [inline docs/contributing/featureflags.md:134]: 💭 Doesn't QA deploy feature branches sometimes? I realize this isn't the norm but unsure if we should mention that `master` branch specifically here.

Comment (trmartin4) [reply docs/contributing/featureflags.md:134]: Good point.  I've updated to remove the reference to `master`.

File: docs/contributing/featureflags.md (hunk 2 above)
Comment (trmartin4) [reply docs/contributing/featureflags.md]: Fixed!  Thank you for catching that.

**Pull request merged**